File: src/main/kotlin/com/wtw/demo/comparator/BooleanDiffer.java
package com.wtw.demo.comparator;

import java.util.Objects;
import java.util.Optional;

public class BooleanDiffer implements Differ<Boolean> {

    @Override
    public <S extends Boolean> Optional<DiffResult> diff(S a, S b) {
        return Objects.equals(a,b)  ?
                Optional.of(DiffResult.change(a, b)) :
                Optional.empty();
    }
}

File: src/main/kotlin/com/wtw/demo/comparator/BooleanDiffer.kt
package com.wtw.demo.comparator

import java.util.*

class BooleanDiffer : Differ<Boolean?> {
    override fun diff(a: Boolean?, b: Boolean?): Optional<DiffResult> {
        return if (a == b) Optional.of(DiffResult.change(a, b)) else Optional.empty()
    }
}
